src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -249,6 +249,301 @@ public Task CollectionRuleOptions_EventCounterTrigger_GreaterThanLargerThanLessT
+        {
+            return ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetAspNetRequestDurationTrigger(options =>
+                        {
+                            options.RequestCount = -1;
+                            options.SlidingWindowDuration = TimeSpan.FromSeconds(-1);
+                            options.RequestDuration = TimeSpan.FromSeconds(-1);
+                        });
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+
+                    Assert.Equal(3, failures.Length);
+                    VerifyRangeMessage<int>(failures, 0, nameof(AspNetRequestDurationOptions.RequestCount), "1", int.MaxValue.ToString());
+                    VerifyRangeMessage<TimeSpan>(failures, 1, nameof(AspNetRequestDurationOptions.RequestDuration),
+                        AspNetRequestDurationOptions.RequestDuration_MinValue, AspNetRequestDurationOptions.RequestDuration_MaxValue);
+                    VerifyRangeMessage<TimeSpan>(failures, 2, nameof(AspNetRequestDurationOptions.SlidingWindowDuration),
+                        TriggerOptionsConstants.SlidingWindowDuration_MinValue, TriggerOptionsConstants.SlidingWindowDuration_MaxValue);
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_AspNetResponseStatusTrigger_MinimumOptions()
+        {
+            const int ExpectedResponseCount = 10;
+            string[] ExpectedStatusCodes = { "400", "500" };
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetAspNetResponseStatusTrigger(options =>
+                        {
+                            options.ResponseCount = ExpectedResponseCount;
+                            options.StatusCodes = ExpectedStatusCodes;
+                        });
+                },
+                ruleOptions =>
+                {
+                    AspNetResponseStatusOptions responseStatusOptions = ruleOptions.VerifyAspNetResponseStatusTrigger();
+                    Assert.Equal(ExpectedResponseCount, responseStatusOptions.ResponseCount);
+                    Assert.Equal(ExpectedStatusCodes, responseStatusOptions.StatusCodes);
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_AspNetResponseStatusTrigger_RoundTrip()
+        {
+            const int ExpectedResponseCount = 10;
+            TimeSpan ExpectedSlidingWindowDuration = TimeSpan.FromSeconds(45);
+            string[] ExpectedIncludePaths = { "IncludePath1", "IncludePath2" };
+            string[] ExpectedExcludePaths = { "ExcludePath1", "ExcludePath2" };
+            string[] ExpectedStatusCodes = { "400", "500" };
```

Comment:
Not sure we have tests for this already, but there is a regex that forces to be valid status code ranges.